pylearn2/space/__init__.py

```diff
@@ -548,6 +576,87 @@ def _format_as_impl(self, is_numeric, batch, target_space):
         raise NotImplementedError("%s does not implement _format_as_impl()." %
                                   type(self))
 
+    def undo_np_format_as(self, batch, space):
+        """
+        Returns a numeric batch (e.g. a numpy.ndarray or scipy.sparse sparse
+        array), with formatting from space undone.
+
+        This is just a wrapper around self._undo_format_as, with an extra check
+        to throw an exception if <batch> is symbolic.
+
+        Parameters
+        ----------
+        batch : numpy.ndarray, or one of the scipy.sparse matrices.
+            Array which lies in this space.
+        space : Space
+            Space to undo formatting from.
+
+        Returns
+        -------
+        WRITEME
```

Comment:
You can simply put the type here, for instance `numpy.ndarray, or one of the scipy.sparse matrices` instead of WRITEME. You do not have to name the returned value.